### ISSUE TITLE
App Platform: Representation of App Versions and Their Metadata in the Control Plane

### ISSUE THREAD

**Issue by cokiengchiara** [giantswarm/roadmap#207]
## Related Epics
- [x] Spec Representation of Catalog Entries (Apps) within the Control Plane https://github.com/giantswarm/roadmap/issues/26

## Milestone 1
- [x] MVP for application versions and metadata representation in the control plane https://github.com/giantswarm/giantswarm/issues/10930 (private repo)

**Subtasks**
- [x] Design AppCatalogEntry CRD https://github.com/giantswarm/apiextensions/pull/572
- [x] Replace dummy with appcatalogentry resource in app-operator appcatalog controller and add empty integration test.
- [x] Implement creating appcatalogentry CRs for the giantswarm catalog with just data from index.yaml for the 1st iteration.

## Milestone 2
- [x] Implement build tool to propagate static app metadata to app catalog https://github.com/giantswarm/giantswarm/issues/10948 (private repo)
- [x] Add error-proofing fields from metadata files https://github.com/giantswarm/giantswarm/issues/13973
* That only one instance of app runs in a cluster clusterSingleton
* That app runs in one namespace namespaceSingleton
* That app runs in a fixed namespace fixedNamespace
* gpuInstances

## Milestone 3
- [x] Validate by checking AppCatalogEntry CRs https://github.com/giantswarm/giantswarm/issues/15852
* name is valid and exists in the catalog
* version is valid and exists in the catalog
## Remaining stories
- [x] Limit the number of AppCatalogEntries to X latest versions of App https://github.com/giantswarm/giantswarm/issues/14800
- [x] Enable AppCatalogEntries for all catalogs and Control Planes https://github.com/giantswarm/giantswarm/issues/14801 -- waiting for agreement on caching
- [x] Implement changes to AppCatalogEntry CR https://github.com/giantswarm/giantswarm/issues/14802

## Background

This is an internal story, upon which important technical requirements and user facing stories depend on.

Currently, an App Catalog is represented in the CP by its appcatalog CR. There's no information in the CP about the App Catalog's contents (Apps). This prevents us from providing "Error-Proofing" functionality, such as:

- Compatibility - is a new app's version compatible with a current cluster's platform release version? is a new platform release compatible with all the current apps?
- App-related cluster limitations - a specific app must be deployed only once per cluster
- Dependencies - this app requires certain features, for example, GPU nodes
- User Requirements - which user values of the App can be left as default and which must be configured by the user, in order to work in the cluster.

**Comment by cokiengchiara** [comment]
Team is making progress on this. Finishing build tool https://github.com/giantswarm/giantswarm/issues/10948 and admission controller https://github.com/giantswarm/giantswarm/issues/11656

**Comment by tomahawk28** [comment]
I don't think `Milestone 3:  Validate by checking AppCatalogEntry CRs` had been not finished, didn't it? 
Also the linked issue, https://github.com/giantswarm/giantswarm/issues/11656 is not relevant, it's not using AppCatalogEntry at all. 
So I created a new issue; https://github.com/giantswarm/giantswarm/issues/15852

**Comment by cokiengchiara** [comment]
> * [ ]  Validate by checking AppCatalogEntry CRs [giantswarm/giantswarm#15852](https://github.com/giantswarm/giantswarm/issues/15852)
> 
> * name is valid and exists in the catalog
> * version is valid and exists in the catalog

thank you @tomahawk28 ❤️  that was my mistake! Edited it now

**Comment by tomahawk28** [comment]
Milestone 3 is done, should we close this roadmap issue as well?

**Comment by cokiengchiara** [comment]
> Milestone 3 is done, should we close this roadmap issue as well?

Yes, but first regarding communications...
1. Docs https://github.com/giantswarm/roadmap/issues/238#issuecomment-795117739
2. Marketing / communications -- I wonder if it makes sense to make an announcement on this... @giantswarm/team-batman @giantswarm/sig-product-marketing 

Key messages (not sure if entirely correct):
* App Platform is built on Helm -- emphasize no vendor lock in https://github.com/giantswarm/giantswarm/issues/15648
* There are restrictions to helm charts... This is how we address it. Is it? 😅 
* Here's our vision for the future...

**Comment by cokiengchiara** [comment]
Implementation is done and released. New issue for communication is here https://github.com/giantswarm/giantswarm/issues/16199 closing.
